Build/Build.cs

```diff
@@ -351,36 +355,110 @@ void ReportTestOutcome(params string[] globFilters)
+    Target InstallNodeWindows => _ => _
+        .OnlyWhenDynamic(() => EnvironmentInfo.IsWin)
+        .Unlisted()
+        .Executes(() =>
         {
-            if (msg.StartsWith("warning"))
+            string os = "win-x64";
+            string archiveType = ".zip";
+
+            var downloadUrl = NodeDownloadUrl.Replace("[OS]", os).Replace("[ARCHIVE_TYPE]", archiveType);
+            var archive = TempDir / $"node.{archiveType}";
+
+            HttpTasks.HttpDownloadFile(downloadUrl, archive);
+
+            var extractDir = NodeDir;
+            var nodeDirPerOs = extractDir / $"node-v{NodeVersion}-{os}";
+
+            archive.UnZipTo(extractDir);
+            Node = ToolResolver.GetTool(nodeDirPerOs / "node.exe");
+            Npm = ToolResolver.GetTool(nodeDirPerOs / "npm.cmd");
+
+            NpmEnvironmentVariables = new()
             {
-                Warning(msg);
-            }
-            else
+                { "PATH", nodeDirPerOs }
+            };
+        });
+
+    Target InstallNodeOsx => _ => _
+        .OnlyWhenDynamic(() => EnvironmentInfo.IsOsx)
+        .Unlisted()
+        .Executes(() =>
+        {
+            string os = "osx-x64";
+            string archiveType = ".tar.gz";
+
+            var downloadUrl = NodeDownloadUrl.Replace("[OS]", os).Replace("[ARCHIVE_TYPE]", archiveType);
+            var archive = TempDir / $"node{archiveType}";
+
+            HttpTasks.HttpDownloadFile(downloadUrl, archive);
+
+            var extractDir = NodeDir;
+            var nodeDirPerOs = extractDir / $"node-v{NodeVersion}-{os}";
+
+            archive.UnTarGZipTo(extractDir);
+            Node = ToolResolver.GetTool(nodeDirPerOs / "bin" / "node");
+            Npm = ToolResolver.GetTool(nodeDirPerOs / "bin" / "npm");
+
+            NpmEnvironmentVariables = new()
             {
-                Error(msg);
-            }
-        }
-    };
+                { "PATH", nodeDirPerOs }
+            };
+        });
+
+    Target InstallNodeLinux => _ => _
+        .OnlyWhenDynamic(() => EnvironmentInfo.IsLinux)
+        .Unlisted()
+        .Executes(() =>
+        {
+            string os = "linux-x64";
+            string archiveType = ".tar.xz";
+
+            var downloadUrl = NodeDownloadUrl.Replace("[OS]", os).Replace("[ARCHIVE_TYPE]", archiveType);
+            var archive = TempDir / $"node{archiveType}";
+
+            HttpTasks.HttpDownloadFile(downloadUrl, archive);
+
+            var extractDir = NodeDir;
+            var nodeDirPerOs = extractDir / $"node-v{NodeVersion}-{os}";
+
+            archive.UnTarGZipTo(extractDir);
+            Node = ToolResolver.GetTool(nodeDirPerOs / "bin" / "node");
+            Npm = ToolResolver.GetTool(nodeDirPerOs / "bin" / "npm");
+
+            NpmEnvironmentVariables = new()
+            {
+                { "PATH", nodeDirPerOs }
+            };
+        });
+
+    string NodeDownloadUrl => $"https://nodejs.org/dist/v{NodeVersion}/node-v{NodeVersion}-[OS][ARCHIVE_TYPE]";
```

Comment:
Why do you use string interpolation for some, but not for others?

Reply:
Not sure if I got you correctly? Because this `string` is a url?

I don't know if a url can also be built with the `/` operator 🤔